src/R/Components/Impl/Controller/IMenuCommand.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+
+namespace Microsoft.R.Components.Controller {
+    /// <summary>
+    /// An object that implements a single command. 
+    /// </summary>
+    public interface IMenuCommand {
```

Comment:
I'd call it `ICommand`, but then it clashes with the existing `ICommand` that is under `Microsoft.Languages.Editor.Controller`. Perhaps the latter should really be `IEditorCommand`? Its members - specifically, `NeedCheckout` - seem to imply that it's meant for that specifically.